doc/lsst.ip.diffim/code_notes.rst

```diff
@@ -0,0 +1,183 @@
+function of the image pixel coordinates.
+
+If ``convolveTemplate==False`` the science exposure is convolved and
+then the template is subtracted from the science image.
+
+The performance of the AL algorithm was studied in details in the
+[Becker_LDM-227]_ report. This study forms the basis of the AL
+algorithm default values; the the degree of the polynomial
+multiplicator of the Gaussian kernel basis functions (``degGauss``),
+the degree of the polynomial that is fitted to the spatial variaton of
+the solution coefficients accross the image (``spatialKernelOrder``)
+and the default detection thresholds (5.5 sigma). As a legacy of this
+study, the ``Winter2013ImageDifferenceTask`` is still available in
+``imageDifference.py`` though it is unclear which test data repository
+it requires.
+
+Due to noise in the template image, convolving the template introduces
+correlation in the noise in the template image. The AL algorithm was
+improved by an additional *afterburner* decorrelation to remove the
+noise correlation in the image difference. The implemented
+decorrelation method and its mathematical formulae of the
+decorrelation kernel is summarised and studied in [Reiss_DMTN-021]_.
+
+Zackay-Ofek-Gal-Yam algorithm
+-----------------------------
+
+[ZOGY2016]_ is free from the assumption that the template is noise
+free or specially selected by any other means. We simply deal with two
+images with different PSFs and noise characteristics (sigma). In the
+basic version of the algorithm, the random noise in the pixels are
+assumed to be background dominated i.e. uncorrelated between pixels
+and independent of the pixel values. Also we assume that the noise has
+zero expectation value i.e. the expectation value of the random noise
+is already removed. ZOGY shows that if these assumptions hold, the
+difference image noise is also independent and identically distributed
+over its pixels (white) by construction, there is no need to
+decorrelate the noise in the difference image.
+
+With the exposure variance planes, exactly the same steps are repeated than on
+the data planes, only the subtraction step is replaced by addition.
+
+The *nan* values are removed from the science and template images
+before Fourier transformations and replaced by the image mean
+values. On the immage difference, the mask plane ``UNMASKEDNAN`` is
+set for pixels where originally any of the two inputs or the
+difference result is *nan*.
+
+Pre-convolution is not implemented in the ZOGY algorithm. If
+``doPreConvolve==True`` the ``S`` the detection likelihood image
+(eq. 12 in [ZOGY2016]_) and its *corrected variance* ``S_var`` (the
+*denominator* of eq. 25 [ZOGY2016]_) are calculated following the
+*simple* correction steps presented in the paper Section 3.3. This
+signal correction is introduced to account for the source noise
+(bright sources) and also for other systematic noise sources. The
+iterative approach of section 3.5 is not implemented.
```

Comment:
This paragraph is a little confusing, because you say pre-convolution is not implemented and then say what happens if doPreConvolve is True. Would it be correct to say that only a simple/linear pre-convolution is implemented and that the recommended/more accurate iterative version is not? Are the limitations of the current implementation understood very well?

Reply:
It's not pre-convolution but a functionality with similar statistical meaning. Clarified.